src/app/tests/TestCommandInteraction.cpp

```diff
@@ -95,8 +123,8 @@ class SimpleTLVPayload : public app::DataModel::EncodableToTLV
     CHIP_ERROR EncodeTo(TLV::TLVWriter & writer, TLV::Tag tag) const override
     {
         TLV::TLVType outerType;
-        ReturnErrorOnFailure(writer.StartContainer(tag, TLV::kTLVType_Structure, outerType));
-        ReturnErrorOnFailure(writer.PutBoolean(chip::TLV::ContextTag(1), true));
+        EXPECT_EQ(writer.StartContainer(tag, TLV::kTLVType_Structure, outerType), CHIP_NO_ERROR);
```

Comment:
Don't you want ASSERT_EQ here? My understanding is that ASSERT ends the test when it is unsafe to proceed further in the test and EXPECT will fail the test but continue running. I am guessing the old reason for `ReturnErrorOnFailure` was because we didn't want the next line of code to run if this like failed.

This comment applies to all instances of the coversions of a `ReturnErrorOnFailure` to and `EXPECT_EQ` in this PR not just here

Reply:
so technically we can not have an `ASSERT_EQ` because this function doesn't return a Void and using ASSERT will make it fail (since the macro will not know what to return) 
However I think we can just leave a ReturnErrorOnFailure as was done before.

Reply:
ReturnErrorOnFailure seems correct. There should be no assert/expect changes here.